client/src/stores/historyItemsStore.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Fetches history items from the API and stores them by `historyId`.
+ * A computed getter returns items for the given `historyId` and
+ * `filterText`.
+ */
+
+import { reverse } from "lodash";
+import { defineStore } from "pinia";
+import Vue, { computed, Ref, ref } from "vue";
+
+import { HistoryFilters } from "@/components/History/HistoryFilters";
+import { mergeArray } from "@/store/historyStore/model/utilities";
+import { LastQueue } from "@/utils/promise-queue";
+import { urlData } from "@/utils/url";
+
+import { type DatasetSummary, type HDCASummary } from "./services";
+
+type HistoryItem = DatasetSummary | HDCASummary;
+
+const limit = 100;
+const queue = new LastQueue();
+
+export const useHistoryItemsStore = defineStore("historyItemsStore", () => {
+    const items: Ref<Record<string, HistoryItem[]>> = ref({});
+    const itemKey = ref("hid");
+    const totalMatchesCount = ref(undefined);
+    const lastCheckedTime = ref(new Date());
+    const lastUpdateTime = ref(new Date());
+    const relatedItems: Ref<Record<string, boolean>> = ref({});
```

Comment:
If I recall correctly, you don't need to import and use `Ref` to define the type, you can simply use:
```suggestion
    const relatedItems = ref<Record<string, boolean>>({});
```

Reply:
Yes, I have use the `Ref` type definition in other places as well, realizing that maybe should've just done this instead, will make that change here though